test/unit/core/type_traits/type_traits_iterator_test.cpp

```diff
@@ -0,0 +1,307 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/std/concepts>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
+#include <list>
+#include <forward_list>
+#include <vector>
+
+#include <seqan3/core/type_traits/iterator.hpp>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/range/views/complement.hpp>
+
+#if SEQAN3_WORKAROUND_GCC_96070
+template <typename base_t>
+struct my_iterator : base_t
+{
+    using difference_type = std::iter_difference_t<base_t>;
+    using value_type = std::iter_value_t<base_t>;
+    using reference = std::iter_reference_t<base_t>;
+    using pointer = void;
+
+    using iterator_category = seqan3::detail::iterator_category_tag_t<base_t>;
+    using iterator_concept = seqan3::detail::iterator_concept_tag_t<base_t>;
+};
+#else // ^^^ workaround / no workaround vvv
+// See https://github.com/seqan/product_backlog/issues/151 how to make this right, depending on how the standard will
+// resolve this issue.
+template <typename base_t>
+struct inherit_iterator_tag : public base_t
+{
+    using iterator_concept = seqan3::detail::iterator_concept_tag_t<base_t>;
+};
+
+template <typename base_t>
+    requires requires()
+    {
+        typename seqan3::detail::iterator_category_tag_t<base_t>;
+    }
```

Comment:
If you move the concept definition up, you can use the concept here.

Reply:
True, but this is currently within a `WORKAROUND` environment. I would need to put it before a lot of code, I can do it, should I?

Reply:
```cpp
template <typename iterator_t>
SEQAN3_CONCEPT iterator_traits_has_iterator_category = requires()
{
    typename std::iterator_traits<iterator_t>::iterator_category;
};

template <typename iterator_t>
SEQAN3_CONCEPT has_iterator_category_tag_t = requires()
{
    typename seqan3::detail::iterator_category_tag_t<iterator_t>;
};
```
is in no if/else block, so it could go to line 18.

Or do I miss something?

Reply:
Done